### DISCUSSION TITLE
Perhaps Reflective Shadow Maps (RMS) could be a good way of implementing Global Illumination in real-time?

### DISCUSSION THREAD

**Discussion by ghost** [godotengine/godot-proposals#9990] [3D]
### Describe the project you are working on

Im working on an fps game with both large and small environments, the game needs to have a stable, high framerate 

### Describe the problem or limitation you are having in your project

The game needs both a good framerate on most hardware and also needs dynamic, accurate lighting
The current GI implementations have drawbacks:
baked GI - isnt dynamic, dynamic objects look like they dont fit in the scene 
voxel GI - has odd bugs (random splotches of darkness) and doesn't work well for the scenes in my project
SDFGI - too slow, it shimmers around when moving and it doesn't provide accurate ambient occlusion

### Describe the feature / enhancement and how it helps to overcome the problem or limitation

I've heard about reflective shadow maps, and i think they would greatly enhance most games made in godot, i am not too sure about RMS's performance impact but i cant imagine it being more expensive than SDFGI, and the quality looks much more desirable, it seems like theres a bit of a shimmer effect as well but i imagine you can filter the result to mitigate 

### Describe how your proposal will work, with code, pseudo-code, mock-ups, and/or diagrams

i am not very knowledgeable on it so i dont know

### If this enhancement will not be used often, can it be worked around with a few lines of script?

i dont think so, if theres a way of doing this within a GDScript then i would like to know

### Is there a reason why this should be core and not an add-on in the asset library?

its core of the lighting system

**Comment by Calinou** [comment]
Moving to discussion, as there is no concrete technical implementation provided.

Note that in general, rendering techniques described in rendering papers tend to suffer from downsides that are unknown until you try to use them in a project in production. I haven't seen too many games out there use reflective shadow maps, so I have my doubts about it.

There's also the issue that Godot already provides 3 different global illumination methods (or even 4 if you consider ReflectionProbes a form of GI). I don't think Godot should offer yet another method, but rather improve or overhaul the existing ones. [Raytracing](https://github.com/godotengine/godot-proposals/discussions/5162) would be an exception here as it allows a level of quality and dynamism that other implementations can't match, but it's too slow to be considered for most projects right now.
